### ISSUE TITLE
jvm fuzzing uploads coverage report to main, not gh-pages

### ISSUE THREAD

**Issue by johannes-riecken** [google/oss-fuzz#7840]
The coverage report (coverage/latest/report/linux) should be uploaded to `storage-repo-branch-coverage`, but gets uploaded to `storage-repo-branch`. Both branches contain other correctly put files.

For reference, see the [two branches of the storage repo](https://github.com/johannes-riecken/pacman-java-storage), which get populated by the [application repo](https://github.com/johannes-riecken/pacman-java).

**Comment by jonathanmetzman** [comment]
It looks to me like your coverage report was uploaded to the gh-pages branch: https://github.com/johannes-riecken/pacman-java-storage/tree/gh-pages/coverage/latest
so the question is why was it also uploaded to main.
As I can see [here](https://github.com/johannes-riecken/pacman-java/runs/6799528957?check_suite_focus=true#step:5:894) your batch fuzzing job is also uploading the coverage report.
My best guess is because you are combining coverage and batch fuzzing in the same job, they are sharing a directory. Note that I didn't expect users to do this. So the options are sperating them into different workflows or maybe I can add some logic to ensure we only upload directories from our workflow run.

**Comment by johannes-riecken** [comment]
Is it envisioned to run like in the Github Actions chapter of the ClusterFuzzLite docs, where I'd set a matrix of `mode: ['batch', 'coverage']` and leave the sanitizer as `address`?
If you compare coverage/latest on the main branch and on the gh-pages branch, then report/linux only appears in main, even though it should be in gh-pages, so that I can view the coverage report at https://johannes-riecken.github.io/pacman-java-storage/coverage/latest/report/linux/report.html (currently dead link).

**Comment by jonathanmetzman** [comment]
>Is it envisioned to run like in the Github Actions chapter of the ClusterFuzzLite docs, where I'd set a matrix of mode: ['batch', 'coverage'] and leave the sanitizer as address?

I don't think you did it the way the docs said (maybe they are unclear). See the doc sections on [coverage](https://google.github.io/clusterfuzzlite/running-clusterfuzzlite/github-actions/#coverage-reports) and [batch fuzzing](https://google.github.io/clusterfuzzlite/running-clusterfuzzlite/github-actions/#batch-fuzzing). Coverage in cflite_cron.yml batch fuzzing goes in cflite_batch.yml.

>If you compare coverage/latest on the main branch and on the gh-pages branch, then report/linux only appears in main, even though it should be in gh-pages, 

I suspect this has something to do with the [last coverage run](https://github.com/johannes-riecken/java-example/runs/6826192057?check_suite_focus=true) failing because of the problem I addressed in your other issue

>so that I can view the coverage report at https://johannes-riecken.github.io/pacman-java-storage/coverage/latest/report/linux/report.html (currently dead link).
